R/render_all.R

```diff
@@ -0,0 +1,96 @@
+#' @title Function to render all protocols into a folder `publish/version_number`.
+#'
+#' @description This function renders all protocols (which have been merged to
+#' the master branch) into a folder `publish/version_number`, where
+#' `version_number` is a tag of the form `YYYY.NN` (year, number giving order of
+#'  release in a year).
+#'
+#'
+#' @param output_root A character string giving the root folder without version
+#' number. Default is "publish".
+#' @param version_number A character string of the form `YYYY.NN` (year, number
+#' giving order of release in a year).
+#'
+#' @return The rendered html files and associated files needed by the html file
+#' for all protocols will be written to subfolders of `publish/version_number`.
+#' The subfolder structure mirrors the structure of `src/thematic` and
+#' `src/project`.
+#'
+#' @importFrom rprojroot find_root is_git_root
+#' @importFrom assertthat assert_that is.string
+#' @importFrom fs dir_ls path path_rel
+#' @importFrom stringr str_detect
+#' @importFrom bookdown render_book
+#' @importFrom purrr pwalk
+#'
+#' @export
+#'
+#'
+#' @examples
+#' \dontrun{
+#' render_all(version_number = "2000.01")
+#'}
+
+
+render_all <- function(output_root = "publish",
+                       version_number = NULL) {
+
+  assert_that(is.string(output_root))
+  assert_that(!is.null(version_number),
+              str_detect(version_number, "^\\d{4}\\.\\d{2}$"),
+              msg = "Please provide a valid version number")
+
+  git_root <- find_root(is_git_root)
+  output_new_root <- path(git_root, output_root, version_number)
```

Comment:
This won't work. The folder `YYYY.NN` should only contain the protocol that was updated during that version. Otherwise the repository would contain too much redundant information.

Reply:
Good point. So all the function needs to do is:

- render the protocol which has that specific `YYYY.NN` tag (possibly more than one in case of multiple languages). The protocol-specific NEWS.Rmd file should reflect which changes have been made.
- automatically update the repo NEWS.Rmd - which gives an overview of all versions of all protocols with links to them and render this NEWS.Rmd into a homepage #19

?

Reply:
Maybe the homepage should only list the procotols, not every version of the protocol. So you could list all protocol folders in `publish`

Reply:
Will `YYYY.NN` effectively become the _git tag_? I ask that because you use the 'tag' term above and also in `render_all()` documentation.

- If so, there is contradiction with the tag names of the branching model, where `protocols-YYYY.NN` is used for clarity. Also, I think that git tags should start with a letter for GitHub to be happy.
- If not, better not call these version numbers 'tags' (but just: version numbers, [see readme](https://github.com/inbo/protocols#version-number)) to prevent confusion.

Reply:
Good catch. Fixed this by 1b96dff